src/main/java/org/aion/wallet/connector/AionTransactionSigner.java

```diff
@@ -0,0 +1,40 @@
+package org.aion.wallet.connector;
+
+import org.aion.api.type.core.tx.AionTransaction;
+import org.aion.crypto.ECKey;
+import org.aion.crypto.ed25519.Ed25519Signature;
+import org.aion.wallet.dto.AccountDTO;
+import org.aion.wallet.dto.AccountType;
+import org.aion.wallet.hardware.HardwareWallet;
+import org.aion.wallet.hardware.HardwareWalletFactory;
+import org.aion.wallet.util.CryptoUtils;
+
+public class AionTransactionSigner {
+
+    private final AccountDTO account;
+
+    public AionTransactionSigner(final AccountDTO from) {
+        account = from;
+    }
+
+    public byte[] sign(final AionTransaction transaction) {
+        final AccountType accountType = account.getType();
+        switch (accountType) {
+            case LOCAL:
+            case IMPORTED:
+                final ECKey ecKey = CryptoUtils.getECKey(account.getPrivateKey());
+                transaction.sign(ecKey);
+                break;
+            case LEDGER:
+            case TREZOR:
+                final HardwareWallet wallet = HardwareWalletFactory.getHardwareWallet(accountType);
```

Comment:
should this code be here? not in ledger?

Reply:
`HardwareWallet` is an interface, the Facatory provides separate implementation for each, according to its type